### PR TITLE
Feat/fpd 50: Login page

### PR DESCRIPTION
**Pull request by SelinaHoHo**
image
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,19 +15,21 @@
     "@commitlint/cli": "^17.4.4",
     "@reduxjs/toolkit": "^2.0.1",
     "@tanstack/react-query": "^5.17.12",
+    "@vee-validate/yup": "^4.12.6",
```
Comment (STUnited-CuongLe) [inline package.json]: ??

File: src/App.tsx [hunk 2]
```diff
@@ -29,6 +30,10 @@ function App() {
   const { theme } = useSelector((state: RootState) => state.global);
   const dispatch = useDispatch();
 
+  // if (localStorage.getItem("isLogin") === "true") {
```
Comment (STUnited-CuongLe) [inline src/App.tsx]: remove comment

File: src/assets/img-signin.jpg
Comment (STUnited-CuongLe) [inline src/assets/img-signin.jpg:1]: ?

File: package.json (hunk 1 above)
Comment (stunited-thanhle) [inline package.json]: ?

File: src/components/protectedRoute/ProtectedRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import { useSelector } from "react-redux";
+
+export const ProtectedRoute = ({ children }: { children: JSX.Element }) => {
+  const { isAuth } = useSelector((state: RootState) => state.auth);
+  if (!isAuth) {
+    // user is n    ot authenticated
+    return <Navigate to='/sign-in' />;
+  }
+  return children;
+};
```
Comment (stunited-thanhle) [inline src/components/protectedRoute/ProtectedRoute.tsx]: remove

File: src/pages/login/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,226 @@
+import { useLogin } from "@/hooks/useAuth";
+import { LockOutlined, MailOutlined } from "@ant-design/icons";
+import { Button, Dropdown, Form, Grid, Input, Tooltip, Typography, theme as antTheme } from "antd";
+import React, { useEffect } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import {} from "../../apis/user.api";
```
Comment (stunited-thanhle) [inline src/pages/login/Login.tsx:7]: remove

File: src/pages/login/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+  const { t } = useTranslation();
+  const [form] = Form.useForm();
+  const { mutate: handleLogin } = useLogin();
+  const { isAuth } = useSelector((state: RootState) => state.auth);
+
+  useEffect(() => {
+    if (isAuth) {
+      navigate("/");
+    }
+  }, [isAuth, navigate]);
```
Comment (stunited-thanhle) [inline src/pages/login/Login.tsx]: Check again, if not, delete it

File: src/pages/login/Login.tsx [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+            >
+              <Input.Password
+                size='large'
+                prefix={
+                  <LockOutlined
+                    style={{ color: "#16c2c2", fontSize: "25px", paddingRight: "5px" }}
+                  />
+                }
+                type='password'
+                placeholder='Password'
```
Comment (stunited-thanhle) [inline src/pages/login/Login.tsx]: Please translate plaeholder

File: src/pages/login/Login.tsx [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+          >
+            <Form.Item
+              name='email'
+              rules={[
+                {
+                  type: "email",
+                  required: true,
+                  message: t("LOGIN.EMAIL_REQUIRED") as string,
+                },
+              ]}
```
Comment (stunited-thanhle) [inline src/pages/login/Login.tsx]: Please use yup for validator. Should be in sync with the team

**Review by stunited-thanhle — APPROVED** [review]

**Pull request merged**